packages/flutter/lib/src/services/system_navigator.dart

```diff
@@ -2,10 +2,60 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:flutter/foundation.dart';
+
 import 'system_channels.dart';
 
 /// Controls specific aspects of the system navigation stack.
 abstract final class SystemNavigator {
+  // This defaults to null because it can't be assumed that the engine is in a
+  // certain state when the app starts. After a hot reload, for example, the
+  // platform will retain its state while this boolean will be reset.
+  static bool? _frameworkHandlesBack;
```

Comment:
This hidden state scares me a little. Why do we need to keep state here that can easily go out of sync with the engine? Why cannot the engine just be the single source of truth?

Reply:
Good point.  I think I was overly cautious of sending duplicate calls of the same value to the engine, but I probably should be anyway, since the engine state could potentially be changed elsewhere.  I'll remove it.